roentgen/absorption/material.py

```diff
@@ -291,17 +291,23 @@ def __init__(self, optical_path, detector):
     def __repr__(self):
         """Returns a human-readable representation."""
         txt = "Response(path="
-        for material in self.optical_path:
-            txt += str(material)
-        txt += " detector=" + str(self.detector)
+        if isinstance(self.optical_path, Material):
+            txt += f"{self.optical_path} "
+        else:  # must be a Stack
+            for material in self.optical_path.materials:
+                txt += f"{material} "
+        txt += f", detector={self.detector})"
```

Comment:
If optical path has to be either a Material or Stack but if they both have reprs could this not be:
```
txt = f"Response(path={self.optical_path},  detector={self.detector})"
```

Reply:
Good point!